create/node.go

```diff
@@ -230,3 +261,51 @@ func getBaseNodeTerraformConfig(terraformModulePath, selectedCluster string, sta
 
 	return cfg, nil
 }
+
+// Returns the hostnames that should be used when adding new nodes. Prevents naming collisions.
+func getNewHostnames(existingNames []string, nodeName string, nodesToAdd int) []string {
+	if nodesToAdd < 1 {
+		return []string{}
+	}
+
+	// If there's only one node to add, and the name doesn't exist
+	// just return the node name itself.
+	if nodesToAdd == 1 {
+		nodeNameUsed := false
+		for _, existingName := range existingNames {
+			if existingName == nodeName {
+				nodeNameUsed = true
+				break
+			}
+		}
+		if !nodeNameUsed {
+			return []string{nodeName}
```

Comment:
Should we still append a `-NUMBER` here? What happens if they add another node later?

Reply:
I was trying to preserve the behavior where the user adds a single node and the hostname matches what the user specified. If they happen to add additional node(s) later those nodes would have the `-NUMBER` suffix resulting in a series of hostnames like `hostname, hostname-1, hostname-2, etc...`. Is it okay if we always add a `-NUMBER` for node hostnames even if they only want to add one node?

Reply:
I think it's okay to always add a `-NUMBER`, even if there's only 1 we'll be consistent. Add it sets us up for good naming when the next node gets added.